### DISCUSSION TITLE
Parent/Child relationships

### DISCUSSION THREAD

**Discussion by paratkd** [jilleb/MQB-FPA#112] [Exploration and research]
Hi there,

Is there any more progress/advice on how the parent/child setups work, and when that might be enabled?

I've had all sorts of fun (not!) on my S3 with all sorts of weird stuff happening; things not showing on the MMI FPA screen, certain modules affecting others, never quite able to set it all up perfectly. And I think this parent/child complexity is a reason why.

Many thanks 👍.

**Comment by jilleb** [comment]
Parent child setups are pretty complex...
a group can have several childs, and each child can have additional request values, outside of the main request value table that has the values for the grouped functions.

The complexity is mostly in the order of the child request values table... I haven't yet found a structured way to parse it nicely... but will do soon.

Ungrouping might help solve your issues.

**Reply by paratkd** [reply]
Thanks for this, appreciate it!

I'll try ungrouping, but I'm sure when I did this last time it didn't work correctly. My MMI seems limited to what can be displayed (particularly thinking of the individual profile where I like to toggle options), but the values do take effect in various modules still.

Is it possible to use the secondary gateway controls to maybe overcome my issues?
For example:

   Option: Interior exhaust sound

   Primary gw control: SAK (int sound control)
   Secondary gw cntrl: AGK (exhaust valves)

Would this actually work, given the control modules are different? 🤔 I'm wanting to send the same Sport (3) request values to both units. Only Interior Soundaktor shows on my MMI.


And, equally, my car combines both engine/dsg gearbox under a 'drive' menu option. The only one that shows up is when I pick Gearbox as a main control. But I want engine set in at the same time (I've viewed live data, and it will also accept 3/sport mode) - so I'm thinking:

Option: Gearbox

   Primary gw control:  GE (trans management) 
   Secondary gw cntrl: MO (engine management)

Not sure if that would work either. These two combinations are my only concern that I haven't been able to nail down. I have AWD, motorway light and steering on my car which all show up on the menu and function fine (but these are simple, seperate options.)

Thanks 👍 🙏

**Reply by paratkd** [reply]
Also-  does the parent/child setup work at all?

For example, I've tried using existing datasets, which have parent/child controls, and then just tried editing those to suit? But I'm not sure if that is even reflected in the output file 🤔

Many thanks.
